tcp/src/server.rs

```diff
@@ -164,19 +163,19 @@ pub struct Server {
 impl Server {
 	/// Closes the server (waits for finish)
 	pub fn close(mut self) {
-		self.stop.take().unwrap().complete(());
+		self.stop.take().map(|sg| sg.send(()).unwrap_or_else(|_| {} /* might be already dropped */));
 		self.remote.take().unwrap().close();
 	}
 
 	/// Wait for the server to finish
 	pub fn wait(mut self) {
-		self.remote.take().unwrap().wait();
+		self.remote.take().map(|handle| handle.wait());
```

Comment:
Why is it possible that handle is not there? Seems that `close` always expects it?
(Would be good to have `.expect` with a proof here and couple of lines above instead of `unwrap`)